save-backend/src/main/kotlin/org/cqfn/save/backend/controllers/UsersDetailsController.kt

```diff
@@ -0,0 +1,54 @@
+package org.cqfn.save.backend.controllers
+
+import org.cqfn.save.backend.repository.UserRepository
+import org.cqfn.save.domain.ImageInfo
+import org.cqfn.save.info.UserInfo
+import org.springframework.security.access.prepost.PreAuthorize
+import org.springframework.web.bind.annotation.*
+import reactor.core.publisher.Mono
+
+/**
+ * Controller that handles operation with users
+ */
+@RestController
+@RequestMapping("/api/users")
+class UsersDetailsController(
+    private val userRepository: UserRepository,
+) {
+    /**
+     * @param userName username
+     * @return [ImageInfo] about user's avatar
+     */
+    @GetMapping("/{userName}/avatar")
+    @PreAuthorize("permitAll()")
+    fun avatar(@PathVariable userName: String): Mono<ImageInfo> = Mono.fromCallable {
+        userRepository.findByName(userName).get().avatar
```

Comment:
You can wrap repository query into our extension method `justOrNotFound` to correctly return 404 instead of throwing in `get()`, if user is absent.

Reply:
Fixed